boefjes/boefjes/plugins/kat_rdns/normalize.py

```diff
@@ -0,0 +1,27 @@
+from typing import Iterable, Union
+
+from dns.message import from_text
+from dns.rdtypes.ANY.PTR import PTR
+
+from boefjes.job_models import NormalizerMeta
+from octopoes.models import OOI, Reference
+from octopoes.models.ooi.dns.records import DNSPTRRecord
+from octopoes.models.ooi.dns.zone import Hostname
+from octopoes.models.ooi.network import Network
+
+
+def run(normalizer_meta: NormalizerMeta, raw: Union[bytes, str]) -> Iterable[OOI]:
```

Comment:
Shouldn't we explicitly handle the "NXDOMAIN" / timeout error here? I think the normalizer works because the first line is ignored and there are no other lines.

Should we also create an object if the reverse DNS doesn't exist? For example for servers that send mail this is important because mail will likely be rejected if there is no reverse DNS.

Reply:
I thought about doing it that way, however, seeing that for many ip's no ptr records are present, this might create a lot of noise.
I'd say a bit which checks for the availability of the PRT record, when the IP is used by a mailserver (or when a mx record, points to a hostname which then resolves into the IP), we'd need a finding as you mentioned.

Reply:
Agree with @underdarknl last point.